examples/receiveWithoutAutoLockRenewAndComplete.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  OnMessage, OnError, MessagingError, delay, ServiceBusMessage, ReceiveMode, Namespace,
+  MessageHandlerOptions
+} from "../lib";
+import * as dotenv from "dotenv";
+import { } from '../lib/streamingReceiver';
+dotenv.config();
+
+const connectionString = "SERVICEBUS_CONNECTION_STRING";
+const entityPath = "QUEUE_NAME";
+const str = process.env[connectionString] || "";
+const path = process.env[entityPath] || "";
+console.log("str: ", str);
+console.log("path: ", path);
+
+let ns: Namespace;
+async function main(): Promise<void> {
+  ns = Namespace.createFromConnectionString(str);
+  const client = ns.createQueueClient(path, { receiveMode: ReceiveMode.peekLock });
+  // Please note: Lock duration property on the Queue was set to 45 seconds.
+  const onMessage: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(">>> Message: ", brokeredMessage);
+    console.log("### Actual message:", brokeredMessage.body ? brokeredMessage.body.toString() : null);
+    const time = 50000;
+    console.log(">>>> Sleeping for %d seconds. Meanwhile autorenew of message lock should NOT happen.", time / 1000);
+    await delay(time);
+    try {
+      await brokeredMessage.complete();
+    } catch (err) {
+      console.log("This should error, since the lock would have expired: %o.", err);
```

Comment:
I was expecting this scenario to throw an error. Servicebus sends the error as seen on line 6 in the debug logs. However, I am not sure how to intercept it.

Looks like I will have to listen for `"settle"` event on the receiver link and resolve/reject the promise based on some information.
Question 1: Not sure where I can find that disposition information in the EventContext for "settle" event in the receiver?
Question 2: When sending disposition for multiple messages, how would I know which disposition was received for which message?

```
  1. rhea:frames [connection-1]:1 -> disposition#15 {"role":true,"state":[]}  +5s
  2. rhea:raw [connection-1] SENT: 28 0000001c02000001005315d00000000c000000054143434200532445 +5s
  3. rhea:io [connection-1] read 366 bytes +5s
  4. rhea:io [connection-1] got frame of size 366 +0ms
  5. rhea:raw [connection-1] RECV: 366 0000016e02000001005315d00000015e0000000642434041005325d00000014d0000000100531dd00000014100000003a31f636f6d2e6d6963726f736f66743a6d6573736167652d6c6f636b2d6c6f7374b100000116546865206c6f636b20737570706c69656420697320696e76616c69642e2045697468657220746865206c6f636b20657870697265642c206f7220746865206d6573736167652068617320616c7265616479206265656e2072656d6f7665642066726f6d207468652071756575652e205265666572656e63653a38383565323561352d646237612d343936312d623432392d6463346536346265396162312c20547261636b696e6749643a39626136306331353030303030303036303034623965326535626439323635645f47385f4232382c2053797374656d547261636b65723a6e6f646573646b3a71756575653a74717e3131312c2054696d657374616d703a31302f33312f3230313820333a35303a343020414d4040 +412ms
  6. rhea:frames [connection-1]:1 <- disposition#15 {"settled":true,"state":[["com.microsoft:message-lock-lost","The lock supplied is invalid. Either the lock expired, or the message has already been removed from the queue. Reference:885e25a5-db7a-4961-b429-dc4e64be9ab1, TrackingId:9ba60c1500000006004b9e2e5bd9265d_G8_B28, SystemTracker:nodesdk:queue:tq~111, Timestamp:10/31/2018 3:50:40 AM",null]]}  +413ms
  7. rhea:events [connection-1] Received disposition for incoming transfers +5s
  8. rhea:events [connection-1] Link got event: settled +0ms
  9. rhea-promise:receiver [connection-1] receiver got event: 'settled'. Re-emitting the translated context. +5s
  10. rhea-promise:translate [connection-1] Translating the context for event: 'settled'. +5s
```